src/libpgmoneta/walfile.c

```diff
@@ -381,4 +383,39 @@ pgmoneta_destroy_walfile(struct walfile* wf)
 
    free(wf->long_phd);
    free(wf);
+}
+
+char*
+pgmoneta_describe_walfile(char* buf, char* path)
+{
+   struct walfile* wf = NULL;
+   struct deque_iterator* record_iterator = NULL;
+   struct configuration* config;
+   struct json* wf_json = NULL;
+
+   config = (struct configuration*)shmem;
+   pgmoneta_json_create(&wf_json);
+
+   pgmoneta_read_walfile(0, path, &wf);
+   if (pgmoneta_deque_iterator_create(wf->records, &record_iterator))
+   {
+      return NULL;
+   }
+
+   while (pgmoneta_deque_iterator_next(record_iterator))
+   {
+      struct json* record_json = NULL;
+      struct decoded_xlog_record* record = (struct decoded_xlog_record*) record_iterator->value->data;
+      pgmoneta_wal_display_decoded_record(&record_json, record, &config->servers[0]);
```

Comment:
If you want to do one line per record here, just print out each of them in FORMAT_JSON_COMPACT here soon as you generate them. No need to put them in one big json and print out all together.